pkg/binding/bindingpolicy-resolution.go

```diff
@@ -36,14 +36,11 @@ type bindingPolicyResolution struct {
 	sync.RWMutex
 
 	// map key is GVK/namespace/name as outputted by util.key.GvkNamespacedNameKey()
-	objectIdentifierToKey map[string]*util.Key
-	destinations          sets.Set[string]
+	objectIdentifierToKey     map[string]*util.Key
+	objectIdentifierToVersion map[string]string // hold resource version for each entry in the above map
```

Comment:
The comment on this type (bindingPolicyResolution) should be updated. That comment currently says

> bindingPolicyResolution stores the selected objects and destinations for a single bindingpolicy.

But this PR is extending what is stored to include a ResoureVersion for each of the selected objects.